internal/node/model/models.go

```diff
@@ -0,0 +1,52 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package model
+
+import (
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type Input struct {
+	// Input index starting from genesis
+	Index uint64 `json:"index"`
+	// Status of the input
+	Status string `json:"status"`
+	// Input data as a blob, starting with '0x'
+	Blob hexutil.Bytes `json:"blob"`
+}
+
+type Output struct {
+	// Input whose processing produced the output
+	InputIndex uint64 `json:"inputIndex"`
+	// Output index within the context of the input that produced it
+	Index uint64 `json:"index"`
+	// Output data as a blob, starting with '0x'
+	Blob hexutil.Bytes `json:"blob"`
+}
+
+type Epoch struct {
+	StartBlock uint64
+	EndBlock   uint64
+}
+
+type InputRange struct {
```

Comment:
I have some doubts about the need to have an InputRange struct. It isn't a "thing" by itself in the architecture, it's a value related to the claim. In your mock, the use of the struct aids in appending the values, but also adds a lot of lines to access these values.

I don't really have an alternative, just food for thought.

Reply:
These are valid points and I agree with you: it isn't needed in any way. 

The `InputRange` struct was a late addition born from my intention to create familiarity between the node's code and the on-chain contracts (at least regarding the main entities) and thus ease understanding. IMO the tradeoff is worth it.